File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        return self.traversal(root, root.val)

    def traversal(self, node, maxVal) -> int:
        if(node is None):
            return 0
        res = None
        if(node.val >= maxVal):
            res = 1
        else:
            res = 0
        maxVal = max(maxVal, node.val)
        res += self.traversal(node.left, maxVal)
        res += self.traversal(node.right, maxVal)

        return res

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    result = 0;
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {
        this.getTreeHeight(root);
        return this.result;
    }
    getTreeHeight(root){
        if(root == null){
            return 0;
        }
        const left = this.getTreeHeight(root.left);
        const right = this.getTreeHeight(root.right);

        this.result = Math.max(this.result, left + right);
        return 1 + Math.max(left, right);
    }
}
